tips/TIP-0045/tip-0045.md

```diff
@@ -0,0 +1,1646 @@
+- Let `Input Score` be `Inputs Count * Work Score Parameters::Input`.
+- Let `Context Input Score` be `Context Inputs Count * Work Score Parameters::Context Input`.
+- Let `Outputs Score` be the sum of work scores of each output `Output` in `Outputs`, where the work score of `Output`
+  is defined as:
+  - Let `Native Token Score` be `Work Score Parameters::Native Token` if `Output` contains a `Native Token Feature`, `0`
+    otherwise.
+  - Let `Token Scheme Score` be `Work Score Parameters::Native Token` if `Output` contains a _Simple Token Scheme_, `0`
+    otherwise.
+  - Let `Staking Score` be `Work Score Parameters::Staking` if `Output` contains a _Staking Feature_, `0` otherwise.
+  - Let `Block Issuer Score` be `Work Score Parameters::Block Issuer` if `Output` contains a _Block Issuer Feature_, `0`
+    otherwise.
+  - Return
+    `Work Score Parameters::Output + Native Token Score + Token Scheme Score + Staking Score + Block Issuer Score`.
+- Let `Allotment Score` be `Work Score Parameters::Allotment * Allotments Count`.
+- Let `Signature Score` be the sum of work scores of the signatures contained in _Signature Unlocks_ within `Unlocks`.
+  - Signature work scores are defined in TIP-46.
+- Return `Size Score + Input Score + Context Input Score + Outputs Score + Allotment Score + Signature Score`.
+
+## Validation
+
+A <i>Transaction Payload</i> has different validation stages since some validation steps can only be executed when
+certain information has (or has not) been received. We therefore distinguish between syntactic and semantic validation.
+
+The different output types and optional output features introduced by [TIP-38](../TIP-0038/tip-0038.md),
+[TIP-1](../TIP-0042/tip-0041.md),[TIP-42](../TIP-0042/tip-0042.md), [TIP-43](../TIP-0043/tip-0043.md) and
+[TIP-44](../TIP-0044/tip-0044.md) add additional constraints to the transaction validation rules, but since these are
+specific to the given outputs and features, they are discussed for each
+[output type](../TIP-0038/tip-0038.md#output-design) and [feature type](../TIP-0038/tip-0038.md#features) separately.
+
+### Syntactic validation
+
+Syntactic validation is checked as soon as the transaction has been received. It validates the structure but not the
+signatures of the transaction. If the transaction does not pass this stage, it must not be broadcast further and can be
+discarded right away.
+
+The following criteria defines whether a payload passes the syntactical validation:
+
+- Transaction:
+  - `Network ID` must match the value of the current network.
+  - Inputs:
+    - `Inputs Count` must be `0 < x ≤ Max Inputs Count`.
+    - For each input the following must be true:
+      - `Input Type` must denote an allowed input type, listed in section [Inputs](#inputs).
+      - `Transaction Output Index` must be `0 ≤ x < Max Outputs Count`.
+    - Each pair of `Transaction ID` and `Transaction Output Index` must be unique in the list of inputs.
+  - Context Inputs:
```

Comment:
Should these be sorted?

Reply:
Since the insertion order of arrays is retained, we don't absolutely need these to be sorted. On the other hand, lexical ordering makes duplicate checking more (memory) efficient, so it seems beneficial to add it. I would do the same for `Inputs` then.